lib/google_drive.rb

```diff
@@ -94,25 +95,15 @@ def self.saved_session(path = nil, proxy = nil, client_id = nil, client_secret =
             "Specifying a proxy object is no longer supported. Set ENV[\"http_proxy\"] instead.")
       end
 
-      if !client_id && !client_secret
-        client_id = "452925651630-egr1f18o96acjjvphpbbd1qlsevkho1d.apps.googleusercontent.com"
-        client_secret = "1U3-Krii5x1oLPrwD5zgn-ry"
-      elsif !client_id || !client_secret
+      if !config.client_id && !config.client_secret
+        config.client_id = "452925651630-egr1f18o96acjjvphpbbd1qlsevkho1d.apps.googleusercontent.com"
+        config.client_secret = "1U3-Krii5x1oLPrwD5zgn-ry"
+      elsif !config.client_id || !config.client_secret
         raise(ArgumentError, "client_id and client_secret must be both specified or both omitted")
       end
 
-      path ||= ENV["HOME"] + "/.ruby_google_drive.token"
-      if ::File.exist?(path)
-        lines = ::File.readlines(path)
-        case lines.size
-          when 1
-            token_data = JSON.parse(lines[0].chomp())
-          when 2
-            # Old format.
-            token_data = nil
-          else
-            raise(ArgumentError, "Not a token file: %s" % path)
-        end
+      if config.access_token_exists?
```

Comment:
Would be fine with:

    token_data = config.access_token

Also, call it refresh_token, neither token_data nor access_token. There is something different called "access token" in OAuth, so calling it access_token is confusing.